node/src/protocol/cryptography.rs

```diff
@@ -163,6 +163,23 @@ impl CryptographicProtocol for ResharingState {
     }
 }
 
+#[async_trait]
+impl CryptographicProtocol for RunningState {
+    async fn progress<C: CryptographicCtx + Send + Sync>(
+        mut self,
+        ctx: C,
+    ) -> Result<NodeState, CryptographicError> {
+        if self.triple_manager.potential_len() < 2 {
+            self.triple_manager.generate();
+        }
+        for (p, msg) in self.triple_manager.poke() {
+            let url = self.participants.get(&p).unwrap();
+            http_client::message(ctx.http_client(), url.clone(), MpcMessage::Triple(msg)).await?;
+        }
```

Comment:
is this correct that we should be doing all the messaging after the protocol completes? What if all nodes are waiting on messages and we'll be stuck in a deadlock?

Reply:
No, the protocol is interactive and occasionally you need to wait for other messages to arrive before you can progress. So ideally there should be a timeout and a restart mechanism but that hasn't been implemented yet.